src/typegoose.ts

```diff
@@ -69,6 +59,23 @@ export class Typegoose {
     return models[name] as ModelType<this> & T;
   }
 
+  public buildPublicSchema<T>(t: T, { existingMongoose, schemaOptions }: GetModelForClassOptions = {}) {
```

Comment:
i would recommend renaming it to just `buildSchema` and to add a jsdoc(/tsdoc) note that it is not needed and automaticly called

-> and the private one `buildPrivateSchema` or `_buildSchema` (and adding a jsdoc(/tsdoc) note that its the private version of it)

Reply:
added